### ISSUE TITLE
[clang-tidy] False positive for cppcoreguidelines-init-variables in Objective-C for-in loop

### ISSUE THREAD

**Issue by brackeen** [llvm/llvm-project#62106]
test.m:
```
#include <Foundation/Foundation.h>
void foo() {
    NSArray<NSString *> *values = @[ ];
    for (NSString *value in values) {
    }
}
```

Running:
`clang-tidy --checks=cppcoreguidelines-init-variables test.m`

```
test.m:4:20: warning: variable 'value' is not initialized [cppcoreguidelines-init-variables]
    for (NSString *value in values) {
                   ^
                         = NULL

```
